src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -290,6 +294,54 @@ protected function getFormActions()
                     ->setUseButtonTag(true)
                     ->addExtraClass('btn-outline-danger btn-hide-outline font-icon-trash-bin action--delete'));
             }
+
+            if ($gridStateStr = ($this->getRequest()->getVar('gridState')
+                || $this->getRequest()->postVar('gridState'))) {
+                $this->gridField->getState(false)->setValue($gridStateStr);
+                $this->owner->gridField->getState(false)->setValue($gridStateStr);
+            }
+
+            $actions->push(HiddenField::create('gridState', null, $this->getRequest()->getVar('gridState')));
+
+            $rightGroup = CompositeField::create()->setName('RightGroup');
+            $rightGroup->addExtraClass('right');
+            $rightGroup->setFieldHolderTemplate(get_class($rightGroup) . '_holder_buttongroup');
+
+            $previousAndNextGroup = CompositeField::create()->setName('PreviousAndNextGroup');
+            $previousAndNextGroup->addExtraClass('rounded');
+            $previousAndNextGroup->setFieldHolderTemplate(get_class($previousAndNextGroup) . '_holder_buttongroup');
+
+            $showPrevious = Config::inst()->get("FormActions", "showPrevious");
+            $showNext = Config::inst()->get("FormActions", "showNext");
+            $showAdd = Config::inst()->get("FormActions", "showAdd");
```

Comment:
If you accept the config changes above, these will need to be refactored as `$this->config()->get()`.